src/app/AppInitGate.tsx

```diff
@@ -1,61 +1,72 @@
 import locales from 'locales'
+import React from 'react'
 import { useAsync } from 'react-async-hook'
+import { Dimensions } from 'react-native'
 import { findBestAvailableLanguage } from 'react-native-localize'
-import { useSelector } from 'react-redux'
+import { useDispatch, useSelector } from 'react-redux'
+import { AppEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
-import { DEFAULT_APP_LANGUAGE } from 'src/config'
-import { initI18n } from 'src/i18n'
-import {
-  allowOtaTranslationsSelector,
-  currentLanguageSelector,
-  otaTranslationsAppVersionSelector,
-} from 'src/i18n/selectors'
+import { appMounted } from 'src/app/actions'
+import { useDeepLinks } from 'src/app/useDeepLinks'
+import i18n from 'src/i18n'
+import { currentLanguageSelector } from 'src/i18n/selectors'
 import useChangeLanguage from 'src/i18n/useChangeLanguage'
 import { navigateToError } from 'src/navigator/NavigationService'
+import { waitUntilSagasFinishLoading } from 'src/redux/sagas'
 import Logger from 'src/utils/Logger'
 
 const TAG = 'AppInitGate'
 
 interface Props {
-  loading: React.ReactNode
+  reactLoadTime: number
+  appStartedMillis: number
   children: React.ReactNode
 }
 
-const AppInitGate = ({ loading, children }: Props) => {
+const AppInitGate = ({ appStartedMillis, reactLoadTime, children }: Props) => {
   const changelanguage = useChangeLanguage()
-  const allowOtaTranslations = useSelector(allowOtaTranslationsSelector)
-  const otaTranslationsAppVersion = useSelector(otaTranslationsAppVersionSelector)
+  const dispatch = useDispatch()
+  const { addDeepLinkListeners } = useDeepLinks()
+
   const language = useSelector(currentLanguageSelector)
   const bestLanguage = findBestAvailableLanguage(Object.keys(locales))?.languageTag
-
-  const i18nInitializer = async () => {
-    await initI18n(
-      language || bestLanguage || DEFAULT_APP_LANGUAGE,
-      allowOtaTranslations,
-      otaTranslationsAppVersion
-    )
-    if (!language && bestLanguage) {
-      await changelanguage(bestLanguage)
-    }
-  }
-
   const initResult = useAsync(
     async () => {
-      Logger.debug(TAG, 'Starting init')
-      await Promise.all([i18nInitializer(), ValoraAnalytics.init()])
-      Logger.debug(TAG, 'init completed')
+      Logger.debug(TAG, 'Starting AppInitGate init')
+      await waitUntilSagasFinishLoading()
+
+      if (!language && bestLanguage) {
+        await changelanguage(bestLanguage)
+      }
```

Comment:
Nothing to do now, but I wonder why we have this here. Shouldn't `initI18n` somehow take care of this?